src/v/tristate.h

```diff
@@ -33,13 +38,16 @@ template<typename T>
 class tristate {
 public:
     using value_type = T;
-    constexpr tristate() noexcept = default;
+    constexpr tristate() noexcept
```

Comment:
* can you explain a bit these changes? they seem like a no-op so why bother?
* why the new constructor with `disable_tristate_t` is needed when the default one does the same thing

Reply:
I think the original issue stems from the fact that 
`tristate<size_t>{}`   could look like a disabled tristate or a `nullopt` tristate, and we default construct it in some critical places that led to this deserialization issue.

changing it everywhere seems not called for, but using a tag type makes it easier to glance over the code, for future reference